### PR TITLE
Updated Stack.java

### LINKED ISSUE #125: Issue in stack
Standard implementation of stack involves only one pointer, which points at top of stack, It means that if you want to view all items of stack, you must remove all elements and then print them, displayStack() function violates the definition of standard stack.

In java also, there is no implementation of such function. However, if you print the stack, then it will print all the elements. But, as far as I know, it is a property of AbstractList class which is super super super class of Stack. I am aware that when we use inheritance, it inherits all the public properties. But, I guess this is unwanted inherited property. Also, If you visit wikipedia link there is no mention of such property of stack.

[https://en.wikipedia.org/wiki/Stack_(abstract_data_type)](url)

### PR DESCRIPTION
**Pull request by Prakash2403**
Fixes #125 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

File: stack/Stack.java [hunk 1]
```diff
@@ -8,76 +8,39 @@
...
-        }
-        for (int i = size-1; i >= 0; i--) {
-                System.out.print(stack.get(i) + " ");
-        }
-        System.out.println();
-    }    
 }
 
-class StackTest {    // Tester Class to check the working of Stack
+public class StackTest {    // Tester Class to check the working of Stack
```
Comment (Monal5031) [inline stack/Stack.java]: File name is `Stack` so the public class name should also be `Stack`

File: stack/Stack.java [hunk 2]
```diff
@@ -8,76 +8,39 @@
...
-        while(obj.getSize() != 0) { // Empty the stack
-            obj.pop();
-        }
-        try {
-        obj.peek(); // Checking for peek() same can be done for pop()
-        } catch (NoSuchElementException exp) {
-            System.out.println("Stack is empty.");
+        System.out.println("\nPoping elements out of stack.");
+        while(true) { // Remove the elements till stack is empty,
+            try{
```
Comment (Monal5031) [inline stack/Stack.java]: Add whitespace after `try`

File: stack/Stack.java [hunk 3]
```diff
@@ -8,76 +8,39 @@
...
-        size++;
     }
-        
-    public T pop() {    // Remove element from top of Stack
-        if (size == 0) {
-            throw new NoSuchElementException("Cannot Pop, Stack is empty");
-        }
-        size--;
+
+    public T pop() throws NoSuchElementException{    // Remove element from top of Stack
```
Comment (Monal5031) [inline stack/Stack.java]: whitespace after `NoSuchElementException`

File: stack/Stack.java [hunk 4]
```diff
@@ -8,76 +8,39 @@
...
-            throw new NoSuchElementException("No element in Stack to Display");
-        }
-        for (int i = size-1; i >= 0; i--) {
-                System.out.print(stack.get(i) + " ");
-        }
-        System.out.println();
-    }    
 }
 
 class StackTest {    // Tester Class to check the working of Stack
```
Comment (singhpratyush) [inline stack/Stack.java]: The filename and public class containing the `main` should be consistent. Please modify accordingly.

I suggest that you add the main method to `Stack` class and self-instantiate there.

**Comment by singhpratyush** [comment]
@Monal5031: Please update your review.

**Pull request merged**